### PR TITLE
Revert new rules & bump the version of PMD / spotbugs and checkstyle

### PR DESCRIPTION
**Pull request by martinvw**
It does require some tests on the existing code bases

### CODE REVIEW

**Comment by wborn** [comment]
Thanks @martinvw! I'll give it a go and see if it fixes the `ShortVariable` warnings for variables names in and exception clauses and lambdas (https://github.com/openhab/static-code-analysis/pull/328#issuecomment-424864404).

**Comment by martinvw** [comment]
Did you give it a go already?

**Comment by wborn** [comment]
Not yet but I think I'll do that this evening!

**Comment by martinvw** [comment]
Never mind than, I will also have some time to do it myself :-)

File: sat-plugin/src/main/java/org/openhab/tools/analysis/tools/SpotBugsChecker.java [hunk 1]
```diff
@@ -82,7 +82,7 @@
     /**
      * The version of the spotbugs-maven-plugin that will be used
      */
-    @Parameter(property = "maven.spotbugs.version", defaultValue = "3.1.5")
+    @Parameter(property = "maven.spotbugs.version", defaultValue = "3.1.7")
     private String spotbugsMavenPluginVersion;
```
Comment (wborn) [inline sat-plugin/src/main/java/org/openhab/tools/analysis/tools/SpotBugsChecker.java]: There's a Spotbugs Maven plugin and Spotbugs itself. 

The most recent released version of: 
* Spotbugs Maven plugin is 3.1.6
* Spotbugs is 3.1.7 (the field below, which also needs to be updated.

Comment (martinvw) [reply sat-plugin/src/main/java/org/openhab/tools/analysis/tools/SpotBugsChecker.java]: This was already out-of-sync before, good that you noticed 👍 

Let me also update the documentation

**Comment by wborn** [comment]
The `ShortVariable` is definitly improved with the new version. But I still see warnings for variables that I think are OK such as 'sb' for StringBuilders and `it` for iterators that are defined outside loops or `b` for byte. WDYT of the results @martinvw?

**Comment by martinvw** [comment]
I'm not such a fan of the examples which seem to remain, so if we can make it the lowest priority I'm fine, can you agree with that? Both a `byte` and a `StringBuilder` also have some meaning and bigger scope. Way more than the `i` in a loop, the `e` in a catch and the `?` in lambda. 

So I personally like the result, @kaikreuzer @maggu2810 @cweitkamp what do you think?

**Comment by wborn** [comment]
Here's an example report: [summary_report.html.gz](https://github.com/openhab/static-code-analysis/files/2425410/summary_report.html.gz)

> make it the lowest priority I'm fine

I always wonder what that exactly means when reading the [Guidelines](https://www.openhab.org/docs/developer/development/guidelines.html#e-logging)

> Please fix all the priority 1 issues and all issues with priority 2 and 3 that are relevant (if you have any doubt don't hesitate to ask).

Is it OK for a contributor to just fix prio 1 and skip 2, 3 all together if he/she doesn't agree with these items?

**Comment by kaikreuzer** [comment]
> So I personally like the result, @kaikreuzer @maggu2810 @cweitkamp what do you think?

Is this the chance to comment on the new checks that have been introduced with #328 (as I had missed to comment there)?

I see that we are adding 1412 warnings for ShortVariable alone - and many of those I would consider false positives, or at least acceptable cases.
When we started the SAT tooling, our aim was to enforce the coding guidelines and potentially have the chance to get the SAT warnings to zero - we wanted to very conservatively add new checks that are in line with the agreed guidelines.
The issue that I see is that we are now adding thousands of warnings that we will never ever get away. This renders those warnings rather useless and they distract the focus from the real issues, so that those are not taken seriously anymore either.

> Is it OK for a contributor to just fix prio 1 and skip 2, 3 all together if he/she doesn't agree with these items?

No, this is not ok. As explained, the idea is to enforce the coding guidelines. So ALL warnings should be fixed and 2+3 priorities are not breaking the build as there can be many cases where you have false positives or at least situations that can be discussed whether the rule should apply or not.

So my clear preference is to only add new checks, if they are clearly helping to enforce the agreed guidelines (which are documented at https://www.eclipse.org/smarthome/documentation/development/guidelines.html) and we stand a chance to keep the overall number of warnings low.

**Comment by martinvw** [comment]
@dstivanov I tested it and it looks all fine to me.

Note that I had to add  https://github.com/openhab/openhab-core/pull/407

**Comment by wborn** [comment]
I'm happy with the other new checks @martinvw! These make the code better and so far I don't see them generating false positives. I had hoped the dependency upgrade had done the same for the `ShortVariable` check.

**Comment by martinvw** [comment]
Local branch is fine, triggering rebuild

**Comment by kaikreuzer** [comment]
Thanks for the ping (although my name is @kaikreuzer ;-)).

Just to recap what we discussed today: It might make sense to clearly define the meaning of the different levels of checks and I think we can find a solution that fits everyones expectation. My suggestion was:
- ERROR: Must be fixed and SAT is sure about it, so it directly fails the build.
- WARN: SAT considers it a violation of our documented guidelines. Those warning must be addressed in order for the PR to be merged. If warnings are still left, there should be a comment, which clearly explains, why this one is an exception from the rule. We must strive to have 0 warnings in our code base in the repo.
- INFO: Code style issues - nothing that really prevents a merge, but which could potentially be addressed to make the code nicer and better readable/maintainable. Nothing to be considered a violation against the documented guidelines, though. It's ok to have INFOs in the code base.

Would you all agree to those?

**Comment by dstivanov** [comment]
@kaikreuzer I agree with one small remark - when a PR build is finished, the warning can be seen only in the logs, but not on the PR. That means that it will not be easily visible for the PR author to fix the issues, but instead s/he will have to wait until a reviewer comes to check the PR. Will it be a good idea to introduce new github PR check for the SAT ?

**Comment by dstivanov** [comment]
@kaikreuzer I agree with one small remark - when a PR build is finished, the warning can be seen only in the logs, but not on the PR. That means that it will not be easily visible for the PR author to fix the issues, but instead s/he will have to wait until a reviewer comes to check the PR. Will it be a good idea to introduce new github PR check for the SAT ?

**Comment by dstivanov** [comment]
@kaikreuzer I agree with one small remark - when a PR build is finished, the warning can be seen only in the logs, but not on the PR. That means that it will not be easily visible for the PR author to fix the issues, but instead s/he will have to wait until a reviewer comes to check the PR. Will it be a good idea to introduce new github PR check for the SAT ?

**Comment by dstivanov** [comment]
@kaikreuzer I agree with one small remark - when a PR build is finished, the WARNs can be seen only in the logs, but not in the PR status. This means that it will not be easily visible for the PR author to see and fix the issues, but instead s/he might wait until a reviewer comes to check the PR. 

Will it be a good idea to introduce new github PR check for the SAT ?

**Comment by kaikreuzer** [comment]
> when a PR build is finished, the WARNs can be seen only in the logs, but not in the PR status.

Our plan was to have the Jenkins PR builds to keep the SAT results as an artifact available and that this should always be linked to when reviewing a PR. If we can somehow even automatically add a comment with the link to these results, it would be imho the best.

> Will it be a good idea to introduce new github PR check for the SAT ?

What exactly do you have in mind here?

**Comment by kaikreuzer** [comment]
> when a PR build is finished, the WARNs can be seen only in the logs, but not in the PR status.

Our plan was to have the Jenkins PR builds to keep the SAT results as an artifact available and that this should always be linked to when reviewing a PR. If we can somehow even automatically add a comment with the link to these results, it would be imho the best.

> Will it be a good idea to introduce new github PR check for the SAT ?

What exactly do you have in mind here?

**Comment by kaikreuzer** [comment]
> when a PR build is finished, the WARNs can be seen only in the logs, but not in the PR status.

Our plan was to have the Jenkins PR builds to keep the SAT results as an artifact available and that this should always be linked to when reviewing a PR. If we can somehow even automatically add a comment with the link to these results, it would be imho the best.

> Will it be a good idea to introduce new github PR check for the SAT ?

What exactly do you have in mind here?

**Comment by kaikreuzer** [comment]
> when a PR build is finished, the WARNs can be seen only in the logs, but not in the PR status.

Our plan was to have the Jenkins PR builds to keep the SAT results as an artifact available and that this should always be linked to when reviewing a PR. If we can somehow even automatically add a comment with the link to these results, it would be imho the best.

> Will it be a good idea to introduce new github PR check for the SAT ?

What exactly do you have in mind here?

**Comment by kaikreuzer** [comment]
> when a PR build is finished, the WARNs can be seen only in the logs, but not in the PR status.

Our plan was to have the Jenkins PR builds to keep the SAT results as an artifact available and that this should always be linked to when reviewing a PR. If we can somehow even automatically add a comment with the link to these results, it would be imho the best.

> Will it be a good idea to introduce new github PR check for the SAT ?

What exactly do you have in mind here?

**Pull request merged**